packages/cli/commands/import/credentials.ts

```diff
@@ -56,60 +74,111 @@ export class ImportCredentialsCommand extends Command {
-					throw new Error(`File does not seem to contain credentials.`);
+				if (!Array.isArray(credentials)) {
+					throw new Error(
+						'File does not seem to contain credentials. Make sure the credentials are contained in an array.',
+					);
 				}
 
-				for (i = 0; i < fileContents.length; i++) {
-					if (typeof fileContents[i].data === 'object') {
+				for (i = 0; i < credentials.length; i++) {
+					if (typeof credentials[i].data === 'object') {
 						// plain data / decrypted input. Should be encrypted first.
-						Credentials.prototype.setData.call(
-							fileContents[i],
-							fileContents[i].data,
-							encryptionKey,
-						);
+						Credentials.prototype.setData.call(credentials[i], credentials[i].data, encryptionKey);
 					}
-					// eslint-disable-next-line no-await-in-loop, @typescript-eslint/no-non-null-assertion
-					await Db.collections.Credentials!.save(fileContents[i]);
+					await this.storeCredential(credentials[i], user);
 				}
 			}
 			console.info(`Successfully imported ${i} ${i === 1 ? 'credential.' : 'credentials.'}`);
-			process.exit(0);
+			process.exit();
 		} catch (error) {
-			console.error('An error occurred while exporting credentials. See log messages for details.');
-			logger.error(error.message);
+			console.error('An error occurred while importing credentials. See log messages for details.');
+			if (error instanceof Error) logger.error(error.message);
 			this.exit(1);
 		}
 	}
+
+	private async initOwnerCredentialRole() {
+		const ownerCredentialRole = await Db.collections.Role!.findOne({
+			where: { name: 'owner', scope: 'credential' },
+		});
+
+		if (!ownerCredentialRole) {
+			throw new Error(`Owner credential role not found. ${FIX_INSTRUCTION}`);
+		}
+
+		this.ownerCredentialRole = ownerCredentialRole;
+	}
+
+	private async storeCredential(credential: object, user: User) {
+		await getConnection().transaction(async (transactionManager) => {
+			const newCredential = new CredentialsEntity();
+
+			Object.assign(newCredential, credential);
+
+			const savedCredential = await transactionManager.save<CredentialsEntity>(newCredential);
+
+			const newSharedCredential = new SharedCredentials();
+
+			Object.assign(newSharedCredential, {
+				credentials: savedCredential,
+				user,
+				role: this.ownerCredentialRole,
+			});
+
+			await transactionManager.save<SharedCredentials>(newSharedCredential);
+		});
+	}
+
+	private async getOwner() {
+		const ownerGlobalRole = await Db.collections.Role!.findOne({
+			where: { name: 'owner', scope: 'global' },
+		});
+
+		const owner = await Db.collections.User!.findOne({ globalRole: ownerGlobalRole });
+
+		if (!owner) {
+			throw new Error(`No owner found. ${FIX_INSTRUCTION}`);
+		}
+
+		return owner;
+	}
+
+	private async getAssignee(id: string) {
+		const user = await Db.collections.User!.findOne(id);
+
+		if (!user) {
+			throw new Error(`Failed to find user with ID ${id}. Are you sure this user exists?`);
```

Comment:
```suggestion
			throw new Error(`Failed to find user with ID ${id}`);
```

Reply:
Then remove also the second sentence in this message?
`'File does not seem to contain workflows. Make sure the workflows are contained in an array.'`

Reply:
Answered in mirror issue